modules/Microsoft.Network/networkInterfaces/deploy.bicep

```diff
@@ -104,7 +104,7 @@ resource networkInterface 'Microsoft.Network/networkInterfaces@2021-05-01' = {
       properties: {
         primary: index == 0 ? true : false
         privateIPAllocationMethod: contains(ipConfiguration, 'privateIPAllocationMethod') ? (!empty(ipConfiguration.privateIPAllocationMethod) ? ipConfiguration.privateIPAllocationMethod : null) : null
-        privateIPAddress: contains(ipConfiguration, 'vmIPAddress') ? (!empty(ipConfiguration.vmIPAddress) ? ipConfiguration.vmIPAddress : null) : null
+        privateIPAddress: contains(ipConfiguration, 'privateIPAddress') ? (!empty(ipConfiguration.privateIPAddress) ? ipConfiguration.privateIPAddress : null) : null
```

Comment:
Looks good. Just one thing: Please also update the test file in the `.test` subfolder - and the module's readme (i.e., deployment examples). This way the test pipelines will continue to work

Reply:
You're right. The change would only apply to my other suggested change, that is, renaming `subnetId` to `subnetResourceId` in alignement with the `publicIPAddressResourceId` property. I'll take care of that in a subsequent PR